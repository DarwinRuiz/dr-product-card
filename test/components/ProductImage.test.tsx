
import React from 'react'
import { render } from '@testing-library/react'
import { ProductImage, } from '../../src/components/ProductImage'
import { ProductCard } from '../../src/components/ProductCard'
import { product1 } from '../data/products'

describe('ProductImage', () => {
    test('renders ProductImage component width custom image succeded', () => {
        const { container } = render(<ProductImage imageUrl="https://hola.jpg" />)

        expect(container).toMatchSnapshot()
    });

    test('Should show the component with product image', () => {
        const { container } = render(<ProductCard product={product1}>
            {
                () => (
                    <ProductImage />
                )
            }
        </ProductCard>)

        expect(container).toMatchSnapshot()
    });
})