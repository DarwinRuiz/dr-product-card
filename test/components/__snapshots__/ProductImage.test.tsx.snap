// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Should show the component with product image 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <img
      alt="Product"
      class="productImg undefined"
      src="public/coffee-mug.png"
    />
  </div>
</div>
`;

exports[`ProductImage renders ProductImage component width custom image succeded 1`] = `
<div>
  <img
    alt="Product"
    class="productImg undefined"
    src="https://hola.jpg"
  />
</div>
`;
